### DISCUSSION TITLE
log4j exploit - any vulnerabilities?

### DISCUSSION THREAD

**Discussion by sspears-deloitte** [reportportal/reportportal#1608] [Q&A] [answered]
Hi. Has anyone confirmed if reportportal is vulnerable to the log4j exploit identified this week?

https://www.lunasec.io/docs/blog/log4j-zero-day/
https://www.govcert.ch/blog/zero-day-exploit-targeting-popular-java-library-log4j/

**Comment by evjlobanova** [comment]
ReportPortal itself uses Logback (http://logback.qos.ch/) as a logging facility.
 
But Log4j used by Elasticsearch 7.10.1. Please, check this (https://discuss.elastic.co/t/zero-day-exploit-in-log4j2-which-is-part-of-elasticsearch/291439/5)  article for details on how to fix Elasticsearch.

**Comment by farazmd** [comment]
Hello, I found these two repositories with 2.13.2 version of log4j , I'm not sure to what extent it is being used, can you confirm if RP is vulnerable because of this ?
service-api
https://github.com/reportportal/service-api/blob/8fb36634c45aaa092c8171cb241fe3377303900a/build.gradle#L137
https://github.com/reportportal/service-api/blob/ca056696f5466fad8706f4c32eb4da8b0f02c907/build.gradle#L64

service-authorization
https://github.com/reportportal/service-authorization/blob/5.6.0/build.gradle#L41

**Reply by DzmitryHumianiuk** [reply]
Updated Log4j to 2.16.0
reportportal/service-authorization#218
reportportal/service-jobs#20
reportportal/service-api#1521

**Reply by farazmd** [reply]
Thank you!

**Comment by DzmitryHumianiuk** [comment, marked as answer]
Please follow here the resolution: https://github.com/reportportal/reportportal/issues/1613
